pkg/kubecost/summaryallocation.go

```diff
@@ -0,0 +1,1155 @@
+package kubecost
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kubecost/cost-model/pkg/log"
+)
+
+// SummaryAllocation summarizes Allocation, keeping only the fields necessary
+// for providing a high-level view of identifying the Allocation (Name) over a
```

Comment:
Anything is possible! This is very abstract, so perhaps we should discuss. If I'm reading between the lines accurately, you could definitely imagine normalizing a lot of our core data, then joining it at advantageous points (query time? cached? whatever).

As an example, we could store only usage and node-ID on `Allocation`, store cost and node-ID on `Node` (i.e. `Asset`) and join to come up with total cost. Reconciliation is only reconciliation if you start with an incorrect price haha. This would just be: compute the price. A very pure idea, indeed. Today, though, we probably survive off of the performance gains from denormalizing, and thus not having to join at query time.

Is that what you have in mind?

Reply:
Yes, my data flow dream basically comes down to normalization and joining. That's a perfect example, and I definitely agree that the denormalization we have today is keeping us alive on performance. My hope would be that "joining it at advantageous points" (i.e. storing raw allocation/asset in durable storage but caching joined data at specified granularity(s)) would counteract the performance hits.